CodeEditModules/Modules/GitClone/src/CheckoutBranchModal.swift

```diff
@@ -0,0 +1,116 @@
+//
+//  CheckoutBranchModal.swift
+//  
+//
+//  Created by Aleksi Puttonen on 14.4.2022.
+//
+
+import Foundation
+import SwiftUI
+import GitClient
+import ShellClient
+
+public struct CheckoutBranchModal: View {
+    private let shellClient: ShellClient
+    @Binding private var isPresented: Bool
+    @Binding private var repoPath: String
+    // TODO: This has to be derived from git
+    @State private var selectedBranch = "main"
+    public init(isPresented: Binding<Bool>,
+                repoPath: Binding<String>,
+                shellClient: ShellClient) {
+        self.shellClient = shellClient
+        self._isPresented = isPresented
+        self._repoPath = repoPath
+    }
+    public var body: some View {
+        VStack(spacing: 8) {
+            HStack {
+                Image(nsImage: NSApp.applicationIconImage)
+                    .resizable()
+                    .frame(width: 64, height: 64)
+                    .padding(.bottom, 50)
+                VStack(alignment: .leading) {
+                    Text("Checkout branch")
+                        .bold()
+                        .padding(.bottom, 2)
+                    Text("Select a branch to checkout")
+                        .font(.system(size: 11))
+                        .foregroundColor(.secondary)
+                        .alignmentGuide(.trailing) { context in
+                        context[.trailing]
+                    }
+                    Menu {
+                        ForEach(getBranches().filter {!$0.contains("HEAD")}, id: \.self) { branch in
+                            Button {
+                                    guard selectedBranch != branch else { return }
+                                    selectedBranch = branch
+                            } label: {
+                                Text(branch)
+                            }.disabled(selectedBranch == branch)
+                        }
+                    } label: {
+                        Text(selectedBranch)
+                    }
+                    HStack {
+                        Button("Cancel") {
+                            isPresented = false
+                        }
+                        Button("Checkout") {
+                            checkoutBranch()
+                        }
+                        .keyboardShortcut(.defaultAction)
+                    }
+                    .alignmentGuide(.trailing) { context in
+                        context[.trailing]
+                    }
+                    .offset(x: 145)
+                }
+            }
+            .padding(.top, 20)
+            .padding(.horizontal, 20)
+            .padding(.bottom, 16)
+            .frame(width: 400)
+        }
+    }
+}
+
+extension CheckoutBranchModal {
```

Comment:
Maybe put this in a separate file... Keep logic away from views at all times.

Reply:
So something along these lines? https://github.com/CodeEditApp/CodeEdit/pull/460/commits/45010c6b459a76491b87904da17fbd5c25a07195

Reply:
If you want you can make it observable but if it does the job it needs to then it all looks good to me